src/NUnitFramework/framework/Constraints/PropertyConstraint.cs

```diff
@@ -70,7 +70,8 @@ public override ConstraintResult ApplyTo<TActual>(TActual actual)
                 throw new ArgumentException(string.Format("Property {0} was not found", name), "name");
 
             propValue = property.GetValue(actual, null);
-            return new ConstraintResult(this, propValue, BaseConstraint.ApplyTo(propValue).IsSuccess);
+            var baseResult = BaseConstraint.ApplyTo(propValue);
+            return new ConstraintResult(this, baseResult.ActualValue, baseResult.IsSuccess);
```

Comment:
This is a minimal fix. A maximal fix, which we could consider as a separate issue, would be for `PropertyConstraint` to use a specially derived type of `ConstraintResult` in order to better tailor its own messages.